internal/partition/partition.go

```diff
@@ -0,0 +1,220 @@
+package partition
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/diskfs/go-diskfs/partition"
+	"github.com/diskfs/go-diskfs/partition/gpt"
+	"github.com/diskfs/go-diskfs/partition/mbr"
+	"github.com/snapcore/snapd/gadget"
+
+	"github.com/canonical/ubuntu-image/internal/helper"
+)
+
+const (
+	// SchemaMBR identifies a Master Boot Record partitioning schema, or an
+	// MBR like role
+	SchemaMBR = "mbr"
+	// SchemaGPT identifies a GUID Partition Table partitioning schema
+	SchemaGPT = "gpt"
+
+	sectorSize512 uint64 = 512
+	sectorSize4k  uint64 = 4096
+
+	protectiveMBRSectors   uint64 = 1
+	partitionHeaderSectors uint64 = 1
+)
+
+// Table is a light wrapper around partition.Table to properly add partitions
+// Some work is sadly duplicated because the go-diskfs lib does not expose
+// the needed data (first/last LBA)
+type Table interface {
```

Comment:
It seems to me that if you added a function that basically does return NewPartitionTable(volume).PartitionTableSize() you wouldn't have to export Table and related things. Would that make sense?

In fact would it make more sense to add GetPartitionTableSize(schema string, sectorSize uint64)?

Reply:
Sadly in Go you cannot define an unexported interface :/. The only way I know to "protect" this interface would be to define it in a nested pkg so that would be more obvious to users this is not supposed to be consumed outside of the `partition` pkg. But it feels a bit overkill.
However, I could unexport NewPartitionTable and create a dedicated function PartitionTableSizeFromVolume to make it clear what is the contract of this pkg.

Reply:
> Sadly in Go you cannot define an unexported interface :/. 

Oh right.

> The only way I know to "protect" this interface would be to define it in a nested pkg so that would be more obvious to users this is not supposed to be consumed outside of the partition pkg. But it feels a bit overkill.

Agree this seems like overkill.

> However, I could unexport NewPartitionTable and create a dedicated function PartitionTableSizeFromVolume to make it clear what is the contract of this pkg.

I think this might help a little.

Reply:
> I think this might help a little.

And that is what I did on Friday.